From ee3d1f09e04643b1b48bd894de1de3c600a01c05 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Thu, 20 Mar 2025 11:57:20 +0100
Subject: [PATCH] ALSA: oxygen: Fix dependency on CONFIG_PM_SLEEP
Git-commit: ee3d1f09e04643b1b48bd894de1de3c600a01c05
Patch-mainline: v6.15-rc1
References: jsc#PED-14430 jsc#PED-14297 jsc#PED-14024

The conversion to EXPORT_SIMPLE_DEV_PM_OPS() also replaced the pm ops
assignment with pm_ptr() macro, but this made difference from the
original code; it  had conditional with ifdef CONFIG_PM_SLEEP, while
we have now with CONFIG_PM, instead.  This seems causing build errors
with randomconfig.

For fixing the inconsistency, replace pm_ptr() with pm_sleep_ptr().

Fixes: 5ea0a2206b58 ("ALSA: oxygen: Convert to EXPORT_SIMPLE_DEV_PM_OPS()")
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202503201853.7kB0BPRw-lkp@intel.com/
Link: https://patch.msgid.link/20250320105721.10789-1-tiwai@suse.de
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/pci/oxygen/oxygen.c   | 2 +-
 sound/pci/oxygen/se6x.c     | 2 +-
 sound/pci/oxygen/virtuoso.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/pci/oxygen/oxygen.c b/sound/pci/oxygen/oxygen.c
index 6d0deea47ee6..ff7439634d76 100644
--- a/sound/pci/oxygen/oxygen.c
+++ b/sound/pci/oxygen/oxygen.c
@@ -855,7 +855,7 @@ static struct pci_driver oxygen_driver = {
 	.id_table = oxygen_ids,
 	.probe = generic_oxygen_probe,
 	.driver = {
-		.pm = pm_ptr(&oxygen_pci_pm),
+		.pm = pm_sleep_ptr(&oxygen_pci_pm),
 	},
 };
 
diff --git a/sound/pci/oxygen/se6x.c b/sound/pci/oxygen/se6x.c
index c1bb9dd1cd54..9d009015d97e 100644
--- a/sound/pci/oxygen/se6x.c
+++ b/sound/pci/oxygen/se6x.c
@@ -138,7 +138,7 @@ static struct pci_driver se6x_driver = {
 	.id_table = se6x_ids,
 	.probe = se6x_probe,
 	.driver = {
-		.pm = pm_ptr(&oxygen_pci_pm),
+		.pm = pm_sleep_ptr(&oxygen_pci_pm),
 	},
 	.shutdown = oxygen_pci_shutdown,
 };
diff --git a/sound/pci/oxygen/virtuoso.c b/sound/pci/oxygen/virtuoso.c
index 0bfe6d8dde3e..ded62199da7f 100644
--- a/sound/pci/oxygen/virtuoso.c
+++ b/sound/pci/oxygen/virtuoso.c
@@ -83,7 +83,7 @@ static struct pci_driver xonar_driver = {
 	.id_table = xonar_ids,
 	.probe = xonar_probe,
 	.driver = {
-		.pm = pm_ptr(&oxygen_pci_pm),
+		.pm = pm_sleep_ptr(&oxygen_pci_pm),
 	},
 	.shutdown = oxygen_pci_shutdown,
 };
-- 
2.52.0

